### PR TITLE
1613 agent timeouts

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes part of  #1613
Improves logging
Fixes firewall rule handling
Changes `keep_tunnel_open` time to 30s from 60s

Max time for agent shutdown:
tunnel timeout = 30
closing http/tcp proxy = 2.5
closing firewall = 1.5
sending log = 5
state telem = 5
quit tunnel = 10

**Total 54s**

Add any further explanations here.

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by running locally, couldn't test tunneling properly due to the bugs in the agent
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/infection_monkey/monkey.py [hunk 1]
```diff
@@ -371,7 +372,8 @@ def cleanup(self):
             self._singleton.unlock()
         except Exception as e:
             logger.error(f"An error occurred while cleaning up the monkey agent: {e}")
-            InfectionMonkey._self_delete()
+            if deleted is None:
+                InfectionMonkey._self_delete()
```
Comment (mssalvatore) [inline monkey/infection_monkey/monkey.py:376]: Deleted may be undefined, resulting in a `NameError`

File: monkey/infection_monkey/network/firewall.py [hunk 2]
```diff
@@ -44,19 +49,23 @@ def __init__(self):
...
+            return None
         except Exception:
             return None
 
+        for line in out.decode().splitlines():
+            if line.startswith("State"):
+                state = line.split()[-1].strip()
+                return state == "ON"
+
+        return None
```
Comment (mssalvatore) [inline monkey/infection_monkey/network/firewall.py:65]: This function is named `is_enabled()`. Should it be returning `bool` instead of `Optional[str]`?

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Our goal was under 30 seconds (120 - 90) and we're currently at 54. To solve this, we could move the `self._monkey_inbound_tunnel.join()` to a `finally`. This would allow all of the other stop logic to run concurrently with closing the tunnel and the maximum stop time would then be 30 seconds.

File: monkey/infection_monkey/monkey.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/infection_monkey/monkey.py:374]: Can't, It's defined in line 346

File: monkey/infection_monkey/network/firewall.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/infection_monkey/network/firewall.py:65]: It is returning `Optional[bool]`

**Comment by VakarisZ** [comment]
Not sure this is a good idea, since if we delete the firewall rules and quit the outbound tunnel the monkeys connecting won't be able to reach the current monkey or island, so what's the point of waiting those 30s?
The ideal case would be to know if machine got exploited and if it connected to the island, I think that would eliminate the 30s wait for the most part. Another option would be to simply NOT wait for exploited machine to join the tunnel. This would save us 30s if monkeys are killed for the price of an arrow showing that monkey communicated back. I think it's worth it.

File: monkey/infection_monkey/network/firewall.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/infection_monkey/network/firewall.py:65]: That's what I mean. Why optional? Why not just return bool?

**Comment by mssalvatore** [comment]
> Not sure this is a good idea, since if we delete the firewall rules and quit the outbound tunnel the monkeys connecting won't be able to reach the current monkey or island, so what's the point of waiting those 30s? The ideal case would be to know if machine got exploited and if it connected to the island, I think that would eliminate the 30s wait for the most part. Another option would be to simply NOT wait for exploited machine to join the tunnel. This would save us 30s if monkeys are killed for the price of an arrow showing that monkey communicated back. I think it's worth it.

Yeah, that's a fair point. We'd have to add extra logic to wait if the stop signal wasn't sent.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**